src/Compilers/Core/Portable/InternalUtilities/ArrayExtensions.cs

```diff
@@ -47,6 +47,29 @@ internal static bool ValueEquals(this uint[] array, uint[] other)
             return true;
         }
 
+        internal static bool ValueEquals(this ulong[] array, ulong[] other)
+        {
+            if (array == other)
+            {
+                return true;
+            }
+
+            if (array == null || other == null || array.Length != other.Length)
+            {
+                return false;
+            }
+
+            for (int i = 0; i < array.Length; i++)
+            {
+                if (array[i] != other[i])
+                {
+                    return false;
+                }
+            }
+
+            return true;
```

Comment:
Consider using Span for this.  IIRC, it has a vector optimized SequenceEquals check.

Reply:
Essentially: `=> array.AsSpan().SequenceEquals(other.AsSpan());`

---
In reply to: [272446240](https://github.com/dotnet/roslyn/pull/34769#discussion_r272446240) [](ancestors = 272446240)